src/components/backup/BackupIcloudStep.js

```diff
@@ -0,0 +1,387 @@
+    } else if (
+      password !== '' &&
+      password.length < 8 &&
+      !passwordRef.current.isFocused()
+    ) {
+      newLabel = 'Use a longer password';
+    } else if (
+      password.length >= 8 &&
+      confirmPassword.length >= 8 &&
+      confirmPassword.length >= password.length &&
+      password !== confirmPassword
+    ) {
+      newLabel = `Passwords don't match`;
+    } else if (password.length >= 8 && !passwordFocused) {
+      newLabel = 'Confirm password';
+    } else if (password.length >= 8 && passwordFocused) {
+      const passInfo = zxcvbn(password);
+      switch (passInfo.score) {
+        case 0:
+        case 1:
+          newLabel = '💩 Weak password';
+          break;
+        case 2:
+          newLabel = '💪 Good password';
+          break;
+        case 3:
+          newLabel = '🦾 Great password';
+          break;
+        case 4:
+          newLabel = '🏰️ Strong password';
+          break;
+        default:
+      }
+    }
+
+    setValidPassword(passwordIsValid);
+    setLabel(newLabel);
+  }, [confirmPassword, loadedPassword, password, passwordFocused]);
+
+  const onPasswordChange = useCallback(
+    ({ nativeEvent: { text: inputText } }) => {
+      setPassword(inputText);
+    },
+    []
+  );
+
+  const onConfirmPasswordChange = useCallback(
+    ({ nativeEvent: { text: inputText } }) => {
+      setConfirmPassword(inputText);
+    },
+    []
+  );
+  const onConfirmBackup = useCallback(async () => {
+    let walletId =
+      params?.walletId ||
+      Object.keys(wallets).find(key => wallets[key].imported === false);
+
+    try {
+      await dispatch(setIsWalletLoading(WalletLoadingStates.BACKING_UP_WALLET));
+
+      let backupFile;
+      if (!latestBackup) {
+        logger.log(
+          'BackupIcloudStep:: backing up to icloud',
+          wallets[walletId]
+        );
+
+        backupFile = await backupWalletToCloud(password, wallets[walletId]);
+      } else {
+        logger.log(
+          'BackupIcloudStep:: adding to icloud backup',
+          wallets[walletId],
+          latestBackup
+        );
+        backupFile = await addWalletToCloudBackup(
+          password,
+          wallets[walletId],
+          latestBackup
+        );
+      }
+      if (backupFile) {
+        logger.log('BackupIcloudStep:: saving backup password');
+        await keychain.saveBackupPassword(password);
+        logger.log('BackupIcloudStep:: saved');
+
+        logger.log('BackupIcloudStep:: backup completed!', backupFile);
+        await dispatch(
+          setWalletBackedUp(walletId, WalletBackupTypes.cloud, backupFile)
+        );
+        logger.log('BackupIcloudStep:: backup saved everywhere!');
+        goBack();
+      } else {
+        Alert.alert('Error while trying to backup');
+        setTimeout(onPasswordSubmit, 1000);
+        dispatch(setIsWalletLoading(null));
+      }
+    } catch (e) {
+      logger.log('Error while backing up', e);
+      dispatch(setIsWalletLoading(null));
```

Comment:
Should we also refocus the input ref (onPasswordSubmit) in the catch as we do in the "else" failure case above?